### PR TITLE
[FIX][MODULAR] Reduces the amount that the S.A.D can be abused.

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
This PR is intended to help make it so that the SAD doesn't function entirely as a free aheal device for medical. I want to accomplish this by doing the following:

- [x] Having organ damage carry over.
- [x] Ensure that brain trauma is carried over.
- [x] Having all regular damage carry over.

This PR isn't going to fix all of the issues the SAD has right now, but it should be a good step in the right direction.

I have also somewhat cleaned up the SAD code and have added autodoc comments to all of the variables. 

## How This Contributes To The Skyrat Roleplay Experience
Medical shouldn't be able to bypass their job mechanics by abusing a machine that is meant as soley a cosmetic device.

## Changelog

:cl:
fix: organ damage now carries over while using the SAD along with brain trauma.
fix: the SAD no longer heals brute and burn damage.
code: cleans up SAD code.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/self_actualization_device/code/self_actualization_device.dm [hunk 1]
```diff
@@ -115,49 +125,112 @@
...
+	var/lung_damage = check_organ(patient, /obj/item/organ/lungs)
+	var/stomach_damage = check_organ(patient, /obj/item/organ/stomach)
+	var/appendix_damage = check_organ(patient, /obj/item/organ/appendix)
+	var/brain_damage = check_organ(patient, /obj/item/organ/brain)
+	var/eye_damage = check_organ(patient, /obj/item/organ/eyes)
+	var/ear_damage = check_organ(patient, /obj/item/organ/ears)
+
+	var/list/trauma_list = list()
+	if(patient.get_traumas())
+		for(var/datum/brain_trauma/trauma in patient.get_traumas())
```
Comment (tf-4) [inline modular_skyrat/modules/self_actualization_device/code/self_actualization_device.dm]:
```suggestion
		for(var/datum/brain_trauma/trauma as anything in patient.get_traumas())
```

File: modular_skyrat/modules/self_actualization_device/code/self_actualization_device.dm [hunk 2]
```diff
@@ -115,49 +125,112 @@
...
+			trauma_list += trauma
+
+	var/brute_damage = patient.getBruteLoss()
+	var/burn_damage = patient.getFireLoss()
+
+	if(obj_flags & EMAGGED)
+		patient.monkeyize()
+
+	else
+		patient?.client?.prefs?.safe_transfer_prefs_to(patient)
```
Comment (tf-4) [inline modular_skyrat/modules/self_actualization_device/code/self_actualization_device.dm]:
```suggestion
		patient.client?.prefs?.safe_transfer_prefs_to(patient)
```
Not really necessary

File: modular_skyrat/modules/self_actualization_device/code/self_actualization_device.dm [hunk 3]
```diff
@@ -115,49 +125,112 @@
...
+
+	open_machine()
+	playsound(src, 'sound/machines/microwave/microwave-end.ogg', 100, FALSE)
+
+/// Checks the damage on the inputed organ and stores it.
+/obj/machinery/self_actualization_device/proc/check_organ(mob/living/carbon/human/patient, obj/item/organ/organ_to_check)
+	var/obj/item/organ/organ_to_track = patient.getorgan(organ_to_check)
+
+	// If the organ is missing, the organ damage is automatically set to 100.
+	if(!organ_to_track)
```
Comment (tf-4) [inline modular_skyrat/modules/self_actualization_device/code/self_actualization_device.dm:201]: Do we really want to give them a new bout of appendicitis if they've had it earlier in the round and their appendix removed due to it? I think there's an argument to be made either way, but I'd lean towards no.

**Comment by softcerv** [comment]
dealing with the wound issue along with languages is going to be moved to future PR so that I can figure out a better solution for them.

**Comment by Iamgoofball** [comment]
Languages theoretically shouldn't need adjustment, if they're properly tied to the mind they should transfer by default.

The healing is probably an unfortunate result of the species swap that's occuring, there's issues upstream with mutation toxins doing heals.

**Comment by softcerv** [comment]
> Languages theoretically shouldn't need adjustment, if they're properly tied to the mind they should transfer by default.
> 
Gotcha, I might have to look and see if there are any runtimes when transferring language prefs.

> The healing is probably an unfortunate result of the species swap that's occuring, there's issues upstream with mutation toxins doing heals.

That would make sense, the healing only seems to apply to brute and burn, along with organs and wounds. I've gotten both of those figured out, but I still need to look into having wounds re-apply.

**Comment by Nerev4r** [comment]
> Languages theoretically shouldn't need adjustment, if they're properly tied to the mind they should transfer by default.

For some reason, they get wiped.

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by Iamgoofball** [comment]
@Nerev4r check if I fucked up and made the languages not apply at the mind level from character creation

**Pull request merged**